### PR TITLE
261 setuptools build fails on upgrade for nixpkgs

### PR DESCRIPTION
**Pull request by eoyilmaz**
- Fixed new-release target in Makefile to do a pull before merging the develop branch.
- Fixed main branch name in Makefile.

### CODE REVIEW

**Comment by eoyilmaz** [comment]
This is not really solving any of the issues related to #261 tbh, but let's merge this

File: Makefile [hunk 1]
```diff
@@ -1,27 +1,40 @@
...
 SETUP_PY_FLAGS = --use-distutils
 VERSION := $(shell cat VERSION_BASE)
 VERSION_FILE=$(CURDIR)/VERSION_BASE
 
 all: build FORCE
 
-build: FORCE
-	python setup.py build -j$(NUM_CPUS) $(SETUP_PY_FLAGS)
+venv:
+	python3.10 -m venv .venv; \
```
Comment (p5k369) [inline Makefile:10]:
```suggestion
	python3 -m venv .venv; \
```

File: Makefile [hunk 2]
```diff
@@ -1,27 +1,40 @@
...
+	source ./.venv/bin/activate; \
+	pip install -r requirements.txt; \
+	pip install -r requirements-dev.txt;
+
+build: venv FORCE
+	source ./.venv/bin/activate; \
+	python -m build;
```
Comment (p5k369) [inline Makefile:17]:
```suggestion
	python3 -m build;
```

**Review by p5k369 — COMMENTED** [review]
I think you're right by setting _pyhton3_ instead of _python _ at line 55, because some distributions still not refer python to python3 even after python2 is obsoloete now. But don't you think we should do that over the whole file? I do not understand why there is an explicit version used for venv creation.

**Review by p5k369 — APPROVED** [review]

**Pull request merged**